engine/runner/src/main/java/org/enso/runner/Main.java

```diff
@@ -880,8 +880,12 @@ private void runMain(
         }
       }
     } catch (PolyglotException e) {
-      printPolyglotException(e, rootPkgPath);
-      throw exitFail();
+      if (e.isExit()) {
```

Comment:
Possibly we might close the context here to notify other language runtimes that the shutdown is happening... not very urgent.